pkg/dockerregistry/server/metrics/metrics.go

```diff
@@ -0,0 +1,47 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Register the metrics.
+func Register() {
+	prometheus.MustRegister(RegistryAPIRequests)
+}
+
+const (
+	registryNamespace = "openshift"
+	registrySubsystem = "registry"
+)
+
+var (
+	RegistryAPIRequests = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Namespace: registryNamespace,
+			Subsystem: registrySubsystem,
+			Name:      "request_duration_microseconds",
+			Help:      "Request latency summary in microseconds for each operation",
+		},
+		[]string{"operation", "name"},
+	)
+)
+
+func NewTimer(collector *prometheus.SummaryVec, labels []string) *metricTimer {
+	return &metricTimer{
+		collector: collector,
+		labels:    labels,
+		startTime: time.Now(),
+	}
+}
+
+type metricTimer struct {
+	collector *prometheus.SummaryVec
+	labels    []string
+	startTime time.Time
+}
+
+func (m *metricTimer) Stop() {
+	m.collector.WithLabelValues(m.labels...).Observe(float64(time.Since(m.startTime) / time.Microsecond))
```

Comment:
I was thinking about logging the error. If the probability of panicking is close to zero, this is ok.